syntax = "proto3";

package game.simenv;

import "protos/types.proto";

service Simenv {
  // 重置仿真服务
  rpc ResetService(types.CommonRequest) returns (types.CommonResponse);
  // 查询仿真服务
  rpc QueryService(types.CommonRequest) returns (types.ServiceState);

  // 获取仿真环境配置
  rpc GetSimenvConfig(types.CommonRequest) returns (SimenvConfig);
  // 设置仿真环境配置
  rpc SetSimenvConfig(SimenvConfig) returns (types.CommonResponse);

  // 控制仿真流程
  rpc SimControl(SimCmd) returns (types.CommonResponse);

  // 获取仿真信息
  rpc SimMonitor(types.CommonRequest) returns (SimInfo);

  // 其他任意调用
  rpc Call(types.CallData) returns (types.CallData);
}

message SimenvConfig {
  string type = 1;  // 仿真环境类型
  string args = 2;  // 仿真环境参数，格式：Json字符串
}

message SimCmd {
  // 预置：init-初始化/start-开始/pause-暂停/step-步进/resume-继续/stop-停止/episode-单局结束/param-设置参数
  string type = 1;    // 仿真命令
  string params = 2;  // 仿真参数，格式：Json字符串
}

message SimInfo {
  // 预置：uninited-未初始化/stopped-已停止/running-正在运行/suspended-已挂起
  string state = 1;  // 仿真状态
  string data = 2;   // 仿真数据，格式：Json字符串
  string logs = 3;   // 仿真日志，格式：Json字符串
}
